ci/Jenkinsfile_utils.groovy

```diff
@@ -150,12 +149,11 @@ def collect_test_results_windows(original_file_name, new_file_name) {
     if (fileExists(original_file_name)) {
         bat 'xcopy ' + original_file_name + ' ' + new_file_name + '*'
         archiveArtifacts artifacts: new_file_name
-        if (env.BRANCH_NAME == "master") {
-          try {
-            s3Upload(file:new_file_name, bucket:env.MXNET_CI_UNITTEST_ARTIFACT_BUCKET, path:utils.get_git_commit_hash().trim()+"-"+env.BUILD_TAG+"/"+new_file_name)
-          } catch (Exception e) {
-            sh "S3 Upload failed ${e}"
-          }
+        try {
+          s3Upload(file:new_file_name, bucket:env.MXNET_CI_UNITTEST_ARTIFACT_BUCKET, path:env.BRANCH_NAME+"-"+utils.get_git_commit_hash().trim()+"-"+env.BUILD_TAG+"/"+new_file_name)
```

Comment:
I'm afraid that the folder starts to get too much context into its name which isn't really scalable.

I'd propose to create a context file (outside of this function since the function is called multiple times per run) where you store all the context you're interested in. The folder names should only look like "PR-12345/22" or "master/852". Everything else like commit hash, target branch, etc should then go into the context file.

Reply:
Sounds good!

Reply:
Resolved by discussion: JOB_NAME/PR/build/file
Because Job-level metrics are required